### PR TITLE
Add initial documentation for SLM

### PR DESCRIPTION
**Pull request by dakrone**
This adds the initial documentation for snapshot lifecycle management.

It also includes the REST spec API json files since they're sort of
documentation.

Relates to #38461

### CODE REVIEW

**Comment by dakrone** [comment]
@elasticmachine run elasticsearch-ci/1

File: docs/reference/ilm/apis/slm-api.asciidoc [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+<1> The policy's version has been incremented because it was updated
+<2> The last successfully initiated snapshot information
+
+It is a good idea to test policies using the execute API to ensure they work.
+
+=== Delete Snapshot Lifecycle Policy API
+
+A policy can be deleted by issuing a delete request with the policy id. Note
+that this prevents any future snapshots from being taken, but does not cancel
+any currently ongoing snapshots.
```
Comment (gwbrown) [inline docs/reference/ilm/apis/slm-api.asciidoc]: Or delete any snapshots previously taken by the policy. Might seem obvious, but probably best to be as clear as possible.

File: docs/reference/ilm/index.asciidoc [hunk 2]
```diff
@@ -47,6 +47,16 @@ to a single shard.
 hardware.
 . Delete the index once the required 30 day retention period is reached.
 
+*Snapshot Lifecycle Management*
+
+ILM itself does allow managing indices, however, outside of the scope of an
+index-level policy is managing snapshots for a set of indices. Since snapshots
```
Comment (gwbrown) [inline docs/reference/ilm/index.asciidoc]: This is a little awkward, I recommend rearranging this just a bit:
```
ILM itself does allow managing indices, however, managing snapshots for a set of indices is
outside of the scope of an index-level policy.
```

**Review by AthenaEryma — CHANGES_REQUESTED** [review]
Great start, just left a few comments about awkward phrasing, etc.

**Comment by dakrone** [comment]
Thanks @gwbrown, I think I addressed all of your comments, thanks for taking a look.

**Review by AthenaEryma — APPROVED** [review]
LGTM, thank you!

**Pull request merged**